### PR TITLE
Add TeamStackPermission resource

### PR DESCRIPTION
**Pull request by mchaynes**
Adds a resource that provisions TeamStackPermissions to this team

### Reviewers:
You can pretty safely ignore anything in the `sdk/` folder, since that is all generated code.

Fixes https://github.com/pulumi/pulumi-pulumiservice/issues/51

### Testing
* New example that is tested automatically on PR builds
* New unit tests for API calls

### CODE REVIEW

**Comment by blampe** [comment]
> You can pretty safely ignore anything in the sdk/ folder, since that is all generated code.

You can automatically hide these from GitHub reviews with gitattributes:
```
$ cat .gitattributes
sdk/* linguist-generated=true
```

**Comment by mchaynes** [comment]
> You can automatically hide these from GitHub reviews with gitattributes

🤯 amazing!

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -104,7 +104,7 @@ jobs:
       fail-fast: true
       matrix:
         dotnetversion: [3.1.301]
-        goversion: [1.17.x]
```
Comment (mchaynes) [inline .github/workflows/main.yml:107]: I needed to update our Go deps because I needed to update our codegen version to use enums.

So I did all that in this PR for simplicity. Happy to break out into a separate PR if that's preferable

**Review by blampe — APPROVED** [review]
